code/API_definitions/connectivity-insights-subscriptions.yaml

```
@@ -0,0 +1,1265 @@
+          description: |
+           'media-type that describes the event payload encoding,
+           must be "application/json" for CAMARA APIs'
+          enum:
+            - application/json
+        data:
+          $ref: "#/components/schemas/NetworkQualityInsight"
+        time:
+          $ref: "#/components/schemas/DateTime"
+      discriminator:
+        propertyName: "type"
+        mapping:
+          org.camaraproject.connectivityinsights.v0.networkQuality-event:
+            "#/components/schemas/NetworkQualityTypeEvent"
+          org.camaraproject.connectivityinsights.v0.eventSubscriptionEnds:
+            "#/components/schemas/EventSubscriptionEnds"
+
+    NetworkQualityTypeEvent:
+      description: event structure for event-type event
+      allOf:
+        - $ref: "#/components/schemas/CloudEvent"
+        - type: object
+          properties:
+            data:
+              $ref: "#/components/schemas/NetworkQualityInsight"
+
+    EventSubscriptionEnds:
+      description: event structure for event subscription ends
+      allOf:
+        - $ref: "#/components/schemas/CloudEvent"
+        - type: object
+          properties:
+            data:
+              $ref: "#/components/schemas/SubscriptionEnds"
+
+    SubscriptionEnds:
+      description: Event detail structure for SUBSCRIPTION_ENDS event
+      type: object
+      required:
+        - terminationReason
+        - subscriptionId
+      properties:
+        terminationReason:
+          $ref: "#/components/schemas/TerminationReason"
+        subscriptionId:
+          $ref: "#/components/schemas/SubscriptionId"
+        terminationDescription:
+          type: string
+
+    TerminationReason:
+      type: string
+      description: |
+        - NETWORK_TERMINATED - API server stopped sending notification
+        - SUBSCRIPTION_EXPIRED - Subscription expire time (optionally set by
+        the requester) has been reached
+        - MAX_EVENTS_REACHED - Maximum number of events (optionally set by the
+        requester) has been reached
+        - ACCESS_TOKEN_EXPIRED - Access Token sinkCredential (optionally set by
+        the requester) expiration time has been reached
+      enum:
+        - MAX_EVENTS_REACHED
+        - NETWORK_TERMINATED
+        - SUBSCRIPTION_EXPIRED
+        - ACCESS_TOKEN_EXPIRED
+
+    DateTime:
+      type: string
+      format: date-time
+      description: |
+        Timestamp of when the occurrence happened. Must adhere to RFC 3339.
+      example: "2018-04-05T17:31:00Z"
+
+    Event-typeNotification:
+      type: string
+      description: |
+        event-type - Event triggered when an event-type event occurred
+        subscription-ends: Event triggered when the subscription ends
+      enum:
+        - org.camaraproject.connectivityinsights.v0.networkQuality-met
+        - org.camaraproject.connectivityinsights.v0.networkQuality-not-met
+        - org.camaraproject.connectivityinsights.v0.subscription-ends
+
```

https://github.com/camaraproject/Commonalities/blob/main/documentation/API-design-guidelines.md#122-event-notification
 For consistency across API we mandate following pattern: `org.camaraproject.<api-name>.<api-version>.<event-name>`
The **type** field is prefixed with a reverse-DNS name, so it is case insensitive, but convention is to use lower case 


```suggestion
      enum:
        - org.camaraproject.connectivity-insights-subscriptions.v0.network-quality-met
        - org.camaraproject.connectivity-insights-subscriptions.v0.network-quality-not-met
        - org.camaraproject.connectivity-insights-subscriptions.v0.subscription-ends
```